skbio/alignment/_indexing.py

```diff
@@ -0,0 +1,206 @@
+            else:
+                return self._slice_on_second_axis(self._obj, indexable)
+
+        if type(indexable) is tuple:
+            if len(indexable) > 2:
+                raise ValueError("Can only slice on two axis. Tuple is length:"
+                                 " %r" % len(indexable))
+            elif len(indexable) > 1:
+                return self._handle_both_axes(*indexable)
+            else:
+                indexable, = indexable
+
+        return self._slice_on_first_axis(self._obj, indexable)
+
+    def _handle_both_axes(self, seq_index, pos_index):
+        seq_index = self._convert_ellipsis(seq_index)
+        pos_index = self._convert_ellipsis(pos_index)
+
+        if not hasattr(seq_index, '__iter__') and (seq_index is Ellipsis or
+                                                   seq_index == slice(None)):
+            # Only slice second axis
+            return self._slice_on_second_axis(self._obj, pos_index)
+        else:
+            r = self._slice_on_first_axis(self._obj, seq_index)
+            if type(r) is self._obj.dtype:
+                # [1, 1] [1, *]
+                return r[pos_index]
+            else:
+                # [*, 1] [*, *]
+                return self._slice_on_second_axis(r, pos_index)
+
+    def _slice_on_second_axis(self, obj, indexable):
+        indexable = self._convert_ellipsis(indexable)
+        if self.is_scalar(indexable, axis=1):
+            # [..., 1]
+            return self._get_position(obj, indexable)
+        else:
+            # [..., *]
+            return self._slice_positions(obj, indexable)
+
+    def _slice_on_first_axis(self, obj, indexable):
+        indexable = self._convert_ellipsis(indexable)
+        if self.is_scalar(indexable, axis=0):
+            # [1]
+            return self._get_sequence(obj, indexable)
+        else:
+            # [*]
+            return self._slice_sequences(obj, indexable)
+
+    def _convert_ellipsis(self, indexable):
+        if indexable is Ellipsis:
+            return slice(None)
+        return indexable
+
+    def _get_position(self, obj, indexable):
+        return obj._get_position_(indexable)
+
+    def _slice_positions(self, obj, indexable):
+        indexable = self._assert_bool_vector_right_size(indexable, axis=1)
+        indexable = self._convert_iterable_of_slices(indexable)
+        return obj._slice_positions_(indexable)
+
+    def _convert_iterable_of_slices(self, indexable):
+        # _assert_bool_vector_right_size will have converted the iterable to
+        # an ndarray if it wasn't yet.
+        if isinstance(indexable, np.ndarray) and indexable.dtype == object:
+            indexable = np.r_[tuple(indexable)]
+
+        return indexable
+
+    def _assert_bool_vector_right_size(self, indexable, axis):
+        if isinstance(indexable, np.ndarray):
+            pass
+        elif hasattr(indexable, '__iter__'):
+            indexable = np.asarray(list(indexable))
```

Comment:
Yeah, generators are possible, which while meaningless to `loc` are meaningful to `iloc`. So to share logic, we need to convert.